subprojects/core/src/main/java/org/gradle/api/internal/AbstractTask.java

```diff
@@ -154,6 +155,7 @@ private AbstractTask(TaskInfo taskInfo) {
         this.publicType = taskInfo.publicType;
         assert project != null;
         assert name != null;
+        NameValidator.validate(name);
```

Comment:
I think it would be better to do this in the task factory, where we do some other validation.